sources/subs/PbeImap.class.php

```diff
@@ -91,14 +91,7 @@ public function __construct()
 		$this->_is_gmail = strpos($this->_hostname, '.gmail.') !== false;
 
 		// I suppose that without this information we can't do anything.
-		if (empty($this->_hostname) || empty($this->_username) || empty($this->_password))
-		{
-			return false;
-		}
-		else
-		{
-			return $this;
-		}
+		return (empty($this->_hostname) || empty($this->_username) || empty($this->_password)) ? false : $this;
```

Comment:
Constructors do not have meaningful return values, anything that is returned from here is discarded

https://scrutinizer-ci.com/g/elkarte/Elkarte/inspections/cb1de7e7-f5f8-49d2-9fad-7efcff3e78ab/issues/files/sources/subs/PbeImap.class.php?status=new&orderField=path&order=asc&honorSelectedPaths=0